src/connector/src/parser/postgres.rs

```diff
@@ -163,48 +164,67 @@ pub fn postgres_row_to_owned_row(row: tokio_postgres::Row, schema: &Schema) -> O
                     }
                 }
                 DataType::Varchar => {
-                    match *row.columns()[i].type_() {
-                        // Since we don't support UUID natively, adapt it to a VARCHAR column
-                        Type::UUID => {
-                            let res = row.try_get::<_, Option<uuid::Uuid>>(i);
-                            match res {
-                                Ok(val) => val.map(|v| ScalarImpl::from(v.to_string())),
-                                Err(err) => {
-                                    if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
-                                        tracing::error!(
-                                            suppressed_count,
-                                            column = name,
-                                            error = %err.as_report(),
-                                            "parse uuid column failed",
-                                        );
-                                    }
-                                    None
+                    if let Kind::Enum(_) = row.columns()[i].type_().kind() {
+                        // enum type needs to be handled separately
+                        let res = row.try_get::<_, Option<EnumParser>>(i);
+                        match res {
+                            Ok(val) => val.map(|v| ScalarImpl::from(v.value)),
+                            Err(err) => {
+                                if let Ok(suppressed_count) = LOG_SUPPERSSER.check() {
+                                    tracing::error!(
+                                        suppressed_count,
+                                        column = name,
+                                        error = %err.as_report(),
```

Comment:
Guess this should be `error = err`.

Reply:
https://github.com/risingwavelabs/risingwave/pull/13763#discussion_r1413467727 
`as_report` is recommended.

Reply:
Oh I misunderstood🥵